tests/test_client_functional.py

```diff
@@ -2947,6 +2947,23 @@
             assert False, "never executed"  # pragma: no cover
 
 
+async def test_aiohttp_request_ctx_manager_not_found_exception_is_dns_specific() -> (
+    None
+):
+    # The error raised should be specific to DNS
+    with pytest.raises(aiohttp.ClientConnectorDNSError):
+        async with aiohttp.request("GET", "http://wrong-dns-name.com"):
+            assert False, "never executed"  # pragma: no cover
+
+
+async def test_aiohttp_request_connector_error_non_dns() -> None:
+    # A non-dns counterpart to test_aiohttp_request_ctx_manager_not_found_exception_is_dns_specific
+    with pytest.raises(aiohttp.ClientConnectorError) as excinfo:
+        async with aiohttp.request("GET", "http://localhost:1/"):
+            assert False, "never executed"  # pragma: no cover
+    assert not isinstance(excinfo.value, aiohttp.ClientConnectorDNSError)
```

Comment:
There used to be comments back when the tool used to be called LGTM. I'm not sure anymore. But I can click the "dismiss alert" button since I can see that coverage is actually reported on this line.

OTOH, why don't you put the exception into `pytest.raises()`. That's what's supposed to be checked anyway.

I think CodeQL understands that `assert False` interrupts the control flow of a test function so it naturally thinks the outer line wouldn't be hit. It'd probably be better to call `pytest.fail()` there instead. I wonder if wrapping such a call into `assert_never()` would let us drop the coverage ignore comment as well...